### PR TITLE
Additional check for not-leader slots in LeadershipSchedule tests

### PR DESCRIPTION
**Pull request by carbolymer**
This change introduces additional check for not-leader slots in the tests.

related discussion: https://input-output-rnd.slack.com/archives/CR599HMFX/p1681424630819669 (internal access only)

### CODE REVIEW

File: cardano-testnet/src/Testnet/Util/Assert.hs [hunk 1]
```diff
@@ -71,30 +71,44 @@ newtype Kind = Kind
...
+      _ -> fail $ "Expected kind was TraceNodeIsLeader, found " <> show kind' <> "instead"
 
 instance FromJSON Kind where
   parseJSON = Aeson.withObject "Kind" $ \v ->
     Kind <$> v .: "kind"
 
-getRelevantLeaderSlots :: FilePath -> Int -> H.PropertyT (ReaderT IntegrationState (ResourceT IO)) [Int]
-getRelevantLeaderSlots poolNodeStdoutFile slotLowerBound = do
+getRelevantSlots :: FilePath -> Int -> H.PropertyT (ReaderT IntegrationState (ResourceT IO)) ([Int], [Int])
+getRelevantSlots poolNodeStdoutFile slotLowerBound = do
```
Comment (carbolymer) [inline cardano-testnet/src/Testnet/Util/Assert.hs:96]: This function returns now all slots seen in the logs. This is useful to double check if we didn't omit any slot.

Comment (newhoggy) [reply cardano-testnet/src/Testnet/Util/Assert.hs:96]: Are `TraceNodeIsLeader` and `TraceNodeNotLeader` definitely the only two `TraceNode` kinds?

Comment (carbolymer) [reply cardano-testnet/src/Testnet/Util/Assert.hs:96]: There's also `TraceNodeCannotForge` - should we check for it as well?

File: cardano-testnet/test/Test/Cli/Alonzo/LeadershipSchedule.hs [hunk 2]
```diff
@@ -505,11 +505,19 @@ hprop_leadershipSchedule = integrationRetryWorkspace 2 "alonzo-leadership-schedu
...
     H.assert $ maxActualSlot >= maxSlotExpected
-    pure someLeaderSlots
+    pure (someLeaderSlots, someNotLeaderSlots)
+
+  H.noteShow_ expectedLeadershipSlotNumbers
+  H.noteShow_ leaderSlots
+  H.noteShow_ notLeaderSlots
+
+  -- Double check that we've seen all slots
+  H.annotate "Slots not seen as TraceNodeIsLeader nor TraceNodeNotLeader"
```
Comment (newhoggy) [inline cardano-testnet/test/Test/Cli/Alonzo/LeadershipSchedule.hs:519]: Prefer `H.note_` to be consistent with the other functions we use.

**Review by newhoggy — APPROVED** [review]

File: cardano-testnet/src/Testnet/Util/Assert.hs [hunk 3]
```diff
@@ -71,30 +71,44 @@ newtype Kind = Kind
...
-  , slot :: Int
-  } deriving (Eq, Show)
-
-instance FromJSON TraceNodeIsLeader where
-  parseJSON = Aeson.withObject "TraceNodeIsLeader" $ \v -> do
-    k <- v .: "val" >>= (.: "kind")
-    if k == "TraceNodeIsLeader"
-    then TraceNodeIsLeader k <$> (v .: "val" >>= (.: "slot"))
-    else fail $ "Expected kind was TraceNodeIsLeader, found " <> show k <> "instead"
+data TraceNode
```
Comment (Jimbo4350) [inline cardano-testnet/src/Testnet/Util/Assert.hs:74]: Why the rename? `TraceNodeIsLeader` was a good name.

File: cardano-testnet/src/Testnet/Util/Assert.hs [hunk 4]
```diff
@@ -71,30 +71,44 @@ newtype Kind = Kind
...
+      "TraceNodeNotLeader" -> TraceNode False kind' <$> slotP
+      _ -> fail $ "Expected kind was TraceNodeIsLeader, found " <> show kind' <> "instead"
 
 instance FromJSON Kind where
   parseJSON = Aeson.withObject "Kind" $ \v ->
     Kind <$> v .: "kind"
 
-getRelevantLeaderSlots :: FilePath -> Int -> H.PropertyT (ReaderT IntegrationState (ResourceT IO)) [Int]
-getRelevantLeaderSlots poolNodeStdoutFile slotLowerBound = do
+getRelevantSlots :: FilePath -> Int -> H.PropertyT (ReaderT IntegrationState (ResourceT IO)) ([Int], [Int])
```
Comment (Jimbo4350) [inline cardano-testnet/src/Testnet/Util/Assert.hs:95]: Returning something like `data LeadershipSlots =  LeadershipSlots { leaderSlots :: [Int], notLeaderSlots :: [Int] }` would make this more readable.

**Pull request merged**